aliyun-ecs/src/test/java/org/jclouds/aliyun/ecs/compute/functions/CleanupServerTest.java

```diff
@@ -0,0 +1,250 @@
+//   public void testApply_FirewallRuleNotInNormalState() throws Exception {
+//      final Server server = serverBuilder.state(State.NORMAL).build();
+//      loadServerExpectations(server);
+//      networkApiExpectations();
+//      natRuleExpectations();
+//      publicIpBlockExpectations();
+//
+//      final FirewallRule firewallRule = firewallRuleBuilder.state(State.FAILED_CHANGE).destination(
+//            FirewallRuleTarget.builder()
+//                  .portList(FirewallRuleTarget.PortList.create("portListId", null, null, null, null)).build()).build();
+//      expect(networkApi.listFirewallRules(jcloudsNetworkDomainId))
+//            .andReturn(PagedIterables.onlyPage(IterableWithMarkers.from(Lists.newArrayList(firewallRule))));
+//
+//      powerOffAndDeleteServerExpectations();
+//      applyAndAssertDeleted();
+//   }
+//
+//   private void firewallRuleAndPortListExpectations() {
+//      final FirewallRule firewallRule = firewallRuleBuilder.state(State.NORMAL).destination(FirewallRuleTarget.builder()
+//            .portList(FirewallRuleTarget.PortList.create("portListId", null, null, null, null)).build()).build();
+//      expect(networkApi.listFirewallRules(jcloudsNetworkDomainId))
+//            .andReturn(PagedIterables.onlyPage(IterableWithMarkers.from(Lists.newArrayList(firewallRule))));
+//      networkApi.deleteFirewallRule(firewallRule.id());
+//      expectLastCall();
+//      networkApi.deletePortList(firewallRule.destination().portList().id());
+//      expectLastCall();
+//   }
+//
+//   private void publicIpBlockExpectations() {
+//      final PublicIpBlock publicIpBlock = publicIpBlockBuilder.state(State.NORMAL).build();
+//      expect(networkApi.listPublicIPv4AddressBlocks(jcloudsNetworkDomainId))
+//            .andReturn(PagedIterables.onlyPage(IterableWithMarkers.from(Lists.newArrayList(publicIpBlock))));
+//      networkApi.removePublicIpBlock(publicIpBlock.id());
+//      expectLastCall();
+//   }
+//
+//   private void natRuleExpectations() {
+//      final NatRule natRule = natRuleBuilder.state(State.NORMAL).build();
+//      expect(networkApi.listNatRules(jcloudsNetworkDomainId))
+//            .andReturn(PagedIterables.onlyPage(IterableWithMarkers.from(Lists.newArrayList(natRule))));
+//      networkApi.deleteNatRule(natRule.id());
+//      expectLastCall();
+//   }
+//
+//   private void loadServerExpectations(Server server) {
+//      expect(api.getServerApi()).andReturn(serverApi).anyTimes();
+//      expect(serverApi.getServer(serverId)).andReturn(server);
+//   }
+//
+//   private void applyWithExpectedErrorMessage(String expectedErrorMessage) {
+//      try {
+//         cleanupServer.apply(serverId);
+//      } catch (IllegalStateException e) {
+//         assertEquals(expectedErrorMessage, e.getMessage());
+//      }
+//   }
+//
+//   private void networkApiExpectations() {
+//      expect(api.getNetworkApi()).andReturn(networkApi);
+//   }
+//
+//}
```

Comment:
Why is all this commented?

Reply:
still working on this, forget to exclude from commit, sorry